sagemaker/local/generate_sagemaker_docker_compose.py

```diff
@@ -18,111 +18,224 @@
 """
```

Comment:
Shall we have separate compose.py for partition and training/inference?

Reply:
My proposed design and implementation here uses one compose file, with different sub-parsers and commands. For example to generate a `training` compose you'd use:


```bash
python local/generate_sagemaker_docker_compose.py --image graphstorm:sm  \
    --graph-name ${GRAPH_NAME} --num-instances ${NUM_INSTANCES} \
    --graph-data-s3 ${DATASET_S3_PATH} --region ${REGION} \
    training \
    --task-type ${TASK_TYPE} --train-yaml-s3 ${TRAIN_YAML_S3} \
    --model-artifact-s3 ${OUTPUT_S3_PATH} $GSF_ARGS
```

See the README for details.

Reply:
Can we do the following?
```
python local/generate_sagemaker_docker_compose.py  training \
    --image graphstorm:sm  \
    --graph-name ${GRAPH_NAME} --num-instances ${NUM_INSTANCES} \
    --graph-data-s3 ${DATASET_S3_PATH} --region ${REGION} \
    --task-type ${TASK_TYPE} --train-yaml-s3 ${TRAIN_YAML_S3} \
    --model-artifact-s3 ${OUTPUT_S3_PATH} $GSF_ARGS
```

Reply:
Done in latest commit.